bundle/src/projects/commercial/modules/fronts-banner-adverts.ts

```diff
@@ -0,0 +1,68 @@
+import { createAdSlot, slotSizeMappings } from '@guardian/commercial-core';
+import fastdom from '../../../lib/fastdom-promise';
+import { addSlot } from './dfp/add-slot';
+
+/**
+ * A fronts banner advert will be inserted above each of the following sections.
+ */
+const sections = [
+	'opinion',
+	'sport',
+	'around-the-world',
+	'lifestyle',
+	'in-pictures',
+];
+
+const insertAdvertAboveSection = async (section: string, advertNum: number) => {
+	await fastdom.measure(() => {
+		const sectionNode = document.querySelector(`section#${section}`);
+		if (!sectionNode) return;
+
+		const ad = createAdSlot('fronts-banner', {
+			name: `fronts-banner-${advertNum}`,
+			classes: 'fronts-banner',
+		});
+
+		const adContainer = document.createElement('div');
+		adContainer.className = 'ad-slot-container fronts-banner-container';
+		adContainer.appendChild(ad);
+
+		void fastdom
+			.mutate(() => {
+				sectionNode.parentElement?.insertBefore(
+					adContainer,
+					sectionNode,
+				);
+			})
+			.then(async () => {
+				await addSlot(ad, false, slotSizeMappings['fronts-banner']);
+			});
+	});
+};
+
+export const init = async (): Promise<void> => {
+	const { switches, tests } = window.guardian.config;
+	if (
+		!switches.frontsBannerAds ||
+		tests?.frontsBannerAdsVariant !== 'variant'
+	) {
+		return Promise.resolve();
+	}
+
+	const {
+		isDotcomRendering,
+		page: { contentType, edition },
+	} = window.guardian.config;
+
+	const isUkNetworkFront =
+		contentType === 'Network Front' && edition === 'UK';
+	if (!isUkNetworkFront || isDotcomRendering) {
+		return Promise.resolve();
+	}
```

Comment:
Could these conditions be combined?

Reply:
Sure. My intention was an early return for those not in the test but it's probably not worth it. Plus, we're never going to just remove the test from this piece of code so it's also not a benefit to separate the conditions out for that reason either. Will amend.